drv/as5048b/as5048b.c

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file
+ * @ingroup     drv_as5048b
+ * @author      Diego Badillo-San-Juan <diego.badillo-san-juan@inria.fr>
+ * @brief       Module for reading the AS5048B rotary encoder.
+ *
+ * @copyright   Inria, 2024
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// Include BSP packages
+#include "gpio.h"
+#include "i2c.h"
+#include "as5048b.h"
+
+//=========================== defines ==========================================
+// Define scl and sda gpio pins
+static const gpio_t scl = { .port = 0, .pin = 10 };
+static const gpio_t sda = { .port = 0, .pin = 9 };
+
+// Hardware settings of pins A1 and A2 define the last two bits of the slave address
+// Set A1 and A2 to GND to get default address 0x40
+#define AS5048B_SLAVE_ADDRESS (0x40)
+#define SLAVE_ADDRESS_REG     (0x15)
+
+// Registers where absolute angle is stored
+// 14 bits in total, 6-13(MSB) and 0-5(LSB) respectively
+#define AS5048B_ANGLE_8MSB (0xFE)
+#define AS5048B_ANGLE_6LSB (0xFF)
+
+// Define Pi
+#ifndef M_PI
+#define M_PI (3.14159265358979323846)
+#endif
+
+//============================== public ========================================
+
+// Initialise I2C communication with rotary encoder
+void as5048b_init(void) {
+    // AS5048B doesn't have a WHO_AM_I register
+    db_i2c_init(&scl, &sda);
```

Comment:
With my previous comment this could be changed like:

```suggestion
    db_i2c_init(&db_scl, &db_sda);
```